types/webassemblyjs__ast/index.d.ts

```diff
@@ -0,0 +1,129 @@
+// Type definitions for webassemblyjs__ast x.x
+// Project: https://github.com/baz/foo (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
+// Definitions by: Mohsen Azimi <https://github.com/mohsen1>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare module "@webassemblyjs/ast" {
+    export function traverse(
+        ast: any,
+        visitor: {
+            ModuleImport?: (p: NodePath<ModuleImport>) => void;
+            ModuleExport?: (p: NodePath<ModuleExport>) => void;
+            Start?: (p: NodePath<Start>) => void;
+            Global?: (p: NodePath<Global>) => void;
+        }
+    ): void;
+    export class NodePath<T> {
+        node: T;
+    }
+    export class Node {}
+    export class Identifier extends Node {
+        value: string;
+    }
+    export class Start extends Node {
+        index: Identifier;
+    }
+    export class ModuleImport extends Node {
+        module: string;
+        descr: {
+            type: string;
+            valtype?: string;
+            id?: Identifier;
+            signature?: Signature;
+        };
+        name: string;
+    }
+    export class ModuleExport extends Node {
+        name: string;
+        descr: ModuleExportDescr;
+    }
+    type Index = Identifier | NumberLiteral;
+    export class ModuleExportDescr extends Node {
+        type: string;
+        exportType: string;
+        id: Index;
+    }
+    export class NumberLiteral extends Node {
+        value: number;
+        raw: string;
+    }
+    export class FloatLiteral extends Node {}
```

Comment:
Shouldn't this have a value? How is it different from `NumberLiteral`?

Reply:
I am moving this from webpack. didn't actually write it. But can improve it for sure

Reply:
From where? I don't see these in `webpack/index.d.ts`?

Reply:
Here https://github.com/webpack/webpack/blob/master/declarations.d.ts